### ISSUE TITLE
notebook analyzing IETF attendance and churn

### ISSUE THREAD

**Issue by sbenthall** [datactive/bigbang#445]
At the IETF 110 HRPC meeting, there's an expressed interest in IETF membership/attendance, with respect to:
 - Are new people are joining ? (Organizational health)
 - Are they being retained?
 - Especially, what about retention of women?

**Comment by sbenthall** [comment]
@npdoty did your notebook do churn?

**Comment by npdoty** [comment]
The work on "tenure" can help with answering research questions about churn, but I don't know if we have results that are this level of detail.

My notebook from 2018 including tenure, number of messages, number of newcomers, etc: https://github.com/npdoty/bigbang/blob/ietf-participation/ietf-participation/ietf_wgs_tenure.ipynb

Also, sodestream folks did a similar analysis for their first blog post, with further inquiry into how early interactions (including whether more junior or senior people) predicted longer tenures:
https://sodestream.github.io/impact-of-early-engagement-on-longevity-of-ietf-participation.html

**Comment by sbenthall** [comment]
So is this issue completed?

**Comment by npdoty** [comment]
I have produced notebooks on attendance and tenure, but they aren't integrated into the main repository yet.

There is also definitely more work to be done on churn and factors that influence retention of a participant (in IETF or any other venue  or mailing list).

So, I propose:
* I make a PR to integrate a notebook demonstrating tenure calculations
* we then close this issue
* we keep a list of open research questions somewhere else (since research questions aren't easily managed/closed as GitHub issues) and that churn and retention of new participants should definitely be on that list
* anyone can open new bigbang issues if there are key features needed for measuring/studying churn

**Comment by sbenthall** [comment]
I agree that GitHub issues are not good ways to keep track of open research questions.

GitHub has a more recent feature, 'Discussions', which are intended for broader, open-ended discussions.

Shall I enable the Discussions feature and we can move research related discussion to there?

**Comment by sbenthall** [comment]
@npdoty An update on this?

**Comment by sbenthall** [comment]
@npdoty Any update on this? It is one of the last blockers for the 0.4 release.

**Comment by npdoty** [comment]
Apologies that I got behind on this for quite a while! Anticipate getting this done by early next week.

**Comment by sbenthall** [comment]
I've connected with @npdoty over this, and the verdict is:

- We'll punt this ticket to the next release
- The attendance work (which is actually more recent) is already merged in, which was the other topic mentioned in the issue:
[https://github.com/datactive/bigbang/blob/main/examples/attendance/IETF%20Attendance.ipynb](https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_datactive_bigbang_blob_main_examples_attendance_IETF-2520Attendance.ipynb&d=DwMFaQ&c=slrrB7dE8n7gBJbeO0g-IQ&r=cHqKC_Ho--XmxkrxXZfw_A&m=es4LHoAfZdcNdabj2HETOe-ixc9d3KqY2lBK4JINz94Xwva618opo-1BlfQGPQAi&s=HxtDf9haK6udCBr9drJzDuCel0rvryOQ9bqXvTNJt64&e=)
- So we should include the attendance piece in the release notes even if #445 isn't completely closed yet because of the tenure/churn stuff.
